docs/rfc/000-External-plugin.md

```diff
@@ -0,0 +1,202 @@
+# External plugin API
+
+## Principles
+
+- It must be absolutely isolated within the Mask Network extension.
+- Developers should be able to develop a working plugin without acquiring "licence" or any form of agreement from the Mask team.
+- It must not harm the data or property (wallet) safety.
+
+## General design
+
+An external plugin should be an isolated website, which means it should not be loaded or installed onto the Mask Network.
+
+When Mask Network detects the metadata of an external plugin, it will try to fetch the manifest and display the content in the payload. 
+This process **must not** involve dynamic code execution.
+
+When the user decides to interact with the external plugin, Mask Network will open a popup window and inject some APIs to it.
+
+It is the same when user wants to use the external plugin when composing.
+
+## User story
+
+### Composing with external plugin
+
+#### Discover the plugin
+
+- If the user has used this plugin before, Mask will display it directly.
+
+- If the user knows the plugin URL, they can manually add it.
+
+- Users can explore plugins in a "external plugin collection", aka Masket Place.
+
+#### Composing
+
+After selecting the plugin, a popup window will appear.
+If Mask Network has no permission to the plugin site, Mask Network will jump to a permission granting page first.
+
+After permissions granted, the plugin page will appear.
+
+User interacts with the plugin.
+
+The plugin sends its payload back to the composition UI and the popup dialog closes.
+
+### Seeing external plugin posted by others
+
+User sees the plugin UI rendered by the Mask Network. Partial info, such as on-chain info, can be displayed directly on the card.
+
+User clicks on the post card to interact with the plugin.
+
+A popup window appears.
+If Mask Network has no permission to the plugin site, 
+Mask Network will jump to a permission granting page first.
+After permissions granted, the plugin page will appear.
+
+User interacts with the plugin.
+
+## Technical details
+
+### Plugin definition
+
+A plugin should be deployed on a static HTTPs URL, for example <https://example.com/my-plugin>, let's call it _base url_.
+
+It should provide a manifest file called "mask-plugin-manifest.json". 
+For the example above, it should be located at <https://example.com/my-plugin/mask-plugin-manifest.json>. 
+It should also be in JSONC (JSON with comment) format.
```

Comment:
may be need new JSON parser, https://mdn.io/JSON.parse not support JSONC
or use webpack pre-handler to JSON format.

Reply:
```js
return x
        .split('\n')
        .filter((x) => !x.match(/^ +\/\//))
        .join('\n')
```

this is enough